### PR TITLE
data edge (labeling & closing) consideration revision

### PR DESCRIPTION
**Pull request by JanghyunJK**
- tagging: @stephen-frank @buechler314 

- Issue: https://github.com/NREL/Wattile/issues/253

- (internal) Teams discussion: [1](https://teams.microsoft.com/l/message/19:caffaa1617874d3aab9ef73c903753f4@thread.skype/1672167296533?tenantId=a0f29d7e-28cd-4f54-8442-7885aee7c080&groupId=a673ac7c-234b-4f92-9636-08629146bfd2&parentMessageId=1672167296533&teamName=Intelligent%20Campus%20Team&channelName=Predictive%20Analytics&createdTime=1672167296533&allowXTenantAccess=false), [2](https://teams.microsoft.com/l/message/19:caffaa1617874d3aab9ef73c903753f4@thread.skype/1672266019109?tenantId=a0f29d7e-28cd-4f54-8442-7885aee7c080&groupId=a673ac7c-234b-4f92-9636-08629146bfd2&parentMessageId=1672266019109&teamName=Intelligent%20Campus%20Team&channelName=Predictive%20Analytics&createdTime=1672266019109&allowXTenantAccess=false), [3](https://teams.microsoft.com/l/message/19:caffaa1617874d3aab9ef73c903753f4@thread.skype/1673017663343?tenantId=a0f29d7e-28cd-4f54-8442-7885aee7c080&groupId=a673ac7c-234b-4f92-9636-08629146bfd2&parentMessageId=1673017663343&teamName=Intelligent%20Campus%20Team&channelName=Predictive%20Analytics&createdTime=1673017663343&allowXTenantAccess=false)

- Summary notes:
  - This is only for alfa. I believe consistent changes have to be made on bravo too.
  - Main issues:
    - pandas `rolling` method wasn't behaving as expected.
    - slicing data differently depending on `bin_label` wasn't considered previously resulting in incorrect input windowing.
  - Solutions here:
    - hard-coding `rolling` methods `closed` argument as `right`.
    - modifying input windowing differently based on `bin_label`.
    - reading data differently based on `bin_label`.

- Some test results (and few more included in internal deck [here](https://nrel.sharepoint.com/:p:/r/sites/ICDashboards/Shared%20Documents/2020%20summer%20internships/[201020_WorkingSlide_Wattile.pptx](https://nrel.sharepoint.com/:p:/r/sites/ICDashboards/Shared%20Documents/2020%20summer%20internships/201020_WorkingSlide_Wattile.pptx?d=w2066441e157547a2be7a9e64f145cf9e&csf=1&web=1&e=tBM9X8&nav=eyJzSWQiOjgyMywiY0lkIjozNzEzNzIwNjExfQ)?d=w2066441e157547a2be7a9e64f145cf9e&csf=1&web=1&e=tBM9X8&nav=eyJzSWQiOjgyMywiY0lkIjozNzEzNzIwNjExfQ)):

image

image

image

image

### CODE REVIEW

**Comment by JanghyunJK** [comment]
@haneslinger test is never ending. could you see if I have setup the parametrization correctly? feel free to commit any quick fixes!

**Comment by haneslinger** [comment]
@JanghyunJK Taking a look now

**Comment by stephen-frank** [comment]
@JanghyunJK once this branch is merged I recommend going ahead with the FTLB test models per #250, assuming you can get them trained before your PTO. That way I can try to get them working in SkySpark while you are on PTO. I can also create the rolling stats tests per #260. Then we can look at both when you're back. Sound good?

**Comment by JanghyunJK** [comment]
> @JanghyunJK once this branch is merged I recommend going ahead with the FTLB test models per #250, assuming you can get them trained before your PTO. That way I can try to get them working in SkySpark while you are on PTO. I can also create the rolling stats tests per #260. Then we can look at both when you're back. Sound good?

that's the plan that I talked with Hannah!

**Comment by JanghyunJK** [comment]
@haneslinger let me know if you're done on your side! I just want to test it with my custom data once last time for both alfa/bravo.

File: tests/unit_tests/models/test_bravo.py [hunk 1]
```diff
@@ -176,7 +172,7 @@ def test_get_prediction_vector_for_time(tmpdir, data_processing_configs):
     window_width_target = pd.Timedelta(
         data_processing_configs["input_output_window"]["window_width_target"]
     )
-    bin_interval = pd.Timedelta(data_processing_configs["resample"]["bin_interval"])
+    bin_interval = pd.Timedelta(configs["data_processing"]["resample"]["bin_interval"])
```
Comment (JanghyunJK) [inline tests/unit_tests/models/test_bravo.py:175]: @haneslinger is this intentional? not using `data_processing_configs`

**Comment by JanghyunJK** [comment]
Did a little side test (1) with my custom data, (2) for alfa/bravo, and (3) some combinations of labeling/closing. All looks good! Once approved, I'll submit jobs on HPC per https://github.com/NREL/Wattile/issues/250. @haneslinger @stephen-frank

File: tests/unit_tests/models/test_bravo.py (hunk 1 above)
Comment (haneslinger) [reply tests/unit_tests/models/test_bravo.py:175]: it is intentional. `data_processing_configs` no longer has bin interval.

**Review by haneslinger — APPROVED** [review]

**Pull request merged**